gt/main.c

```diff
@@ -104,12 +106,19 @@ gt_parse_incoming_pkt(struct rte_mbuf *pkt, struct gt_packet_headers *info)
 			return -1;
 
 		outer_ipv6_hdr = (struct ipv6_hdr *)info->outer_l3_hdr;
-		parsed_len += sizeof(struct ipv6_hdr);
+		outer_ipv6_hdr_len = ipv6_skip_exthdr(outer_ipv6_hdr,
+			pkt->data_len - parsed_len, &encasulated_proto);
+                if (outer_ipv6_hdr_len < 0) {
+                        RTE_LOG(ERR, GATEKEEPER,
+                                "gt: failed to parse the IPv6 packet's extension headers!\n");
```

Comment:
This log message is the same for the outer and inner header, we must notice if we are talking about the outer or inner header in the message.